pygraylog/graylogapi.py

```diff
@@ -4,6 +4,11 @@
 from urlparse import urlparse
 class GraylogAPI(object):
     def __init__(self, url, username=None, password=None, api_key=None):
+
+        # check if the username / password vs. api_key cannot be provided at same time
+        if username is not None and api_key is not None:
```

Comment:
This check is raising an error in the unit tests. My question is: should it not matter if the username, password and api_key are set? We could just prefer the API key if it is provided, which is what seems like you added on line 42.